srv/src/main/java/my/bookshop/handlers/SubscriptionHandler.java

```diff
@@ -0,0 +1,30 @@
+package my.bookshop.handlers;
+import org.springframework.stereotype.Component;
+
+import com.sap.cds.services.handler.EventHandler;
+import com.sap.cds.services.handler.annotations.Before;
+import com.sap.cds.services.handler.annotations.ServiceName;
+import com.sap.cds.services.mt.MtAsyncUnsubscribeEventContext;
+import com.sap.cds.services.mt.MtSubscriptionService;
+import com.sap.cds.services.mt.MtUnsubscribeEventContext;
+
+	/**
+	 * Handler that implements subscription logic
+	 */
+	@Component
+	@ServiceName(MtSubscriptionService.DEFAULT_NAME)
+	public class SubscriptionHandler implements EventHandler {
+
+		@Before(event = MtSubscriptionService.EVENT_UNSUBSCRIBE)
+		public void beforeUnsubscribe(MtUnsubscribeEventContext context) {
+			// always delete the subscription
+			context.setDelete(true);
+		}
+
+		@Before(event = MtSubscriptionService.EVENT_ASYNC_UNSUBSCRIBE)
```

Comment:
EVENT_ASYNC_UNSUBSCRIBE is not a public feature of SaaS Registry, thus we shouldn't show it here...

Reply:
I will remove the anynch handling then.